src/Type/Comment/CommentType.php

```diff
@@ -77,15 +77,27 @@ private static function fields() {
 							return ! empty( $comment->comment_post_ID ) ? get_post( $comment->comment_post_ID ) : null;
 						},
 					],
+					/*
 					'author' => [
 						'type' => Types::user(),
 						'description' => __( 'The post field for comments matches the post id the comment is assigned to. This field is equivalent to WP_Comment->comment_post_ID and the value matching the `comment_post_ID` column in SQL.', 'wp-graphql' ),
 						'resolve' => function( \WP_Comment $comment, $args, AppContext $context, ResolveInfo $info ) {
 							$author = new \WP_User( $comment->user_id );
+							return ! empty( $author ) ? $author : null;
+						},
+					],
+					*/
+					// TODO
 
+					'author' => [
+						'type' => Types::comment_author(),
```

Comment:
`type` should be `Types::comment_author_union(),`
`description` could be changed to `The author of the comment`

Then the `resolve` function can be updated to something like: 

```
'resolve' => function( \WP_Comment $comment, $args, AppContext $context, ResolveInfo $info ) {
    
    // Here we determine if the Author is a User or not so we know what to pass through to the Union for resolving
    if ( ! empty( $comment->user_ID ) ) {
       $author = new \WP_User( $comment->user_ID )
    } else if ( ! empty( $comment ) ) {
       $author = $comment;
    } else { 
       $author = null;
    }
    return $author;
}
```